### PR TITLE
[meshcop] publish the full timestamp tlv

### PR DESCRIPTION
**Pull request by wgtdkp**

### CODE REVIEW

**Review by bukepo — APPROVED** [review]

**Review by superwhd — APPROVED** [review]
LGTM :+1:

File: src/border_agent/border_agent.cpp [hunk 1]
```diff
@@ -252,8 +251,14 @@ void BorderAgent::PublishMeshCopService(void)
         }
         else
         {
-            activeTimestamp = htobe64(activeDataset.mActiveTimestamp);
-            txtList.emplace_back("at", reinterpret_cast<uint8_t *>(&activeTimestamp), sizeof(activeTimestamp));
+            uint64_t activeTimestampValue;
+
+            activeTimestampValue = (activeDataset.mActiveTimestamp.mSeconds << 16) |
```
Comment (simonlingoogle) [inline src/border_agent/border_agent.cpp:256]: nit: we can elide the bit operations if `otTimestamp` is defined as an union with a `uint64_t` value to get the full timesstamp.

Comment (wgtdkp) [reply src/border_agent/border_agent.cpp:256]: ditto~

Comment (simonlingoogle) [inline src/border_agent/border_agent.cpp:256]: nit: define constant for 16 and 1.

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Review by librasungirl — APPROVED** [review]

File: src/border_agent/border_agent.cpp (hunk 1 above)
Comment (jwhui) [reply src/border_agent/border_agent.cpp:256]: @wgtdkp , thoughts?

Comment (bukepo) [reply src/border_agent/border_agent.cpp:256]: I think it's fine to use the numbers here. It should be easy to understand.

Comment (wgtdkp) [reply src/border_agent/border_agent.cpp]: I tend to not define constants too far away from where they are being used, especially when there are a single use case. To help readability, I made a short function for converting the timestamp to uint64_t and added documentation for its layout.

**Review by jwhui — APPROVED** [review]

**Pull request merged**